subsys/metrics/src/main/java/org/commonjava/indy/metrics/sigar/FilesystemMetricsSet.java

```diff
@@ -0,0 +1,138 @@
+
+package org.commonjava.indy.metrics.sigar;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricSet;
+import org.hyperic.sigar.FileSystem;
+import org.hyperic.sigar.FileSystemUsage;
+import org.hyperic.sigar.Sigar;
+import org.hyperic.sigar.SigarException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FilesystemMetricsSet implements MetricSet
+{
+    @Override
+    public Map<String, Metric> getMetrics() {
+        final Map<String, Metric> gauges = new HashMap<String, Metric>();
+        Sigar sigar = SigarService.getSigar();
+        try {
+            FileSystem[] fileSystems = sigar.getFileSystemList();
+            for (FileSystem fs : fileSystems) {
+                if (fs.getType() != FileSystem.TYPE_LOCAL_DISK)
+                    continue;
+                createGaugesForFilesystem(gauges, fs);
+            }
+        } catch (SigarException e) {
+            //
```

Comment:
We probably need to at least log this somewhere. Ignoring exceptions will lead to VERY difficult debugging if we have a problem in a deployment later.

Even better would be if there was a way to re-throw this, so we can decide at a higher level whether to proceed with Indy startup if the metrics subsystem isn't working correctly.